### PR TITLE
LPS-199315 DM Share option is now a dropdown with Share and Copy Link options

### PR DESCRIPTION
**Pull request by ambrinchaudhary**
See story https://liferay.atlassian.net/browse/LPS-197479. 

In this pull, in DM we pass **from** 
DM - Share - Before

**to:** 

DM - Share - AFTER

For now Copy Link option does nothing, that will be done in task https://liferay.atlassian.net/browse/LPS-199513

### CODE REVIEW

**Comment by ambrinchaudhary** [comment]
ci:test:sf

File: modules/apps/sharing/sharing-document-library/src/main/java/com/liferay/sharing/document/library/internal/display/context/SharingDLViewFileVersionDisplayContext.java [hunk 1]
```diff
@@ -138,18 +138,46 @@ else if (Objects.equals(
 			i++;
 		}
 
+		DropdownItem sharingDropdownItem =
```
Comment (AliciaGarciaGarcia) [inline modules/apps/sharing/sharing-document-library/src/main/java/com/liferay/sharing/document/library/internal/display/context/SharingDLViewFileVersionDisplayContext.java:141]: Suggestion:

To buy space and to make easier the deletion of the FF, make this instead this code

```
if (FeatureFlagManagerUtil.isEnabled("LPS-197477")) {
			UnsafeConsumer<DropdownContextItem, Exception> sharingActionItem =
				_sharingDropdownItemFactory.createShareActionUnsafeConsumer(
					DLFileEntryConstants.getClassName(),
					_fileEntry.getFileEntryId(), _httpServletRequest);

			if (i >= dropdownItems.size()) {
				dropdownItems.addAll(
					DropdownItemListBuilder.addContext(
						sharingActionItem
					).build());
			}
			else {
				dropdownItems.addAll(
					i,
					DropdownItemListBuilder.addContext(
						sharingActionItem
					).build());
			}
		}
		else {
			DropdownItem sharingDropdownItem =
				_sharingDropdownItemFactory.createShareDropdownItem(
					DLFileEntryConstants.getClassName(),
					_fileEntry.getFileEntryId(), _httpServletRequest);

			if (i >= dropdownItems.size()) {
				dropdownItems.add(sharingDropdownItem);
			}
			else {
				dropdownItems.add(i, sharingDropdownItem);
			}
		}
```

File: modules/apps/sharing/sharing-document-library/src/main/java/com/liferay/sharing/document/library/internal/display/context/SharingDLViewFileVersionDisplayContext.java [hunk 2]
```diff
@@ -174,8 +201,27 @@ else if (Objects.equals(
...
+			_sharingDropdownItemFactory.createShareDropdownItem(
+				DLFileEntryConstants.getClassName(),
+				_fileEntry.getFileEntryId(), _httpServletRequest);
+
+		UnsafeConsumer<DropdownContextItem, Exception> sharingActionItem =
+			_sharingDropdownItemFactory.createShareActionUnsafeConsumer(
+				DLFileEntryConstants.getClassName(),
+				_fileEntry.getFileEntryId(), _httpServletRequest);
+
 		if (i < dropdownItems.size()) {
```
Comment (AliciaGarciaGarcia) [inline modules/apps/sharing/sharing-document-library/src/main/java/com/liferay/sharing/document/library/internal/display/context/SharingDLViewFileVersionDisplayContext.java:214]: same here than the comment above.

make the FF be the principal if, so this way you can declare only or sharingDropdownItem or sharingActionItem and the removal of the FF is easier.

File: modules/apps/sharing/sharing-document-library/src/main/java/com/liferay/sharing/document/library/internal/display/context/SharingDLViewFileVersionDisplayContext.java (hunk 1 above)
Comment (ambrinchaudhary) [reply modules/apps/sharing/sharing-document-library/src/main/java/com/liferay/sharing/document/library/internal/display/context/SharingDLViewFileVersionDisplayContext.java:141]: Done!

File: modules/apps/sharing/sharing-document-library/src/main/java/com/liferay/sharing/document/library/internal/display/context/SharingDLViewFileVersionDisplayContext.java (hunk 2 above)
Comment (ambrinchaudhary) [reply modules/apps/sharing/sharing-document-library/src/main/java/com/liferay/sharing/document/library/internal/display/context/SharingDLViewFileVersionDisplayContext.java:214]: Done!

**Comment by ambrinchaudhary** [comment]
ci:test:sf

**Review by AliciaGarciaGarcia — APPROVED** [review]

**Review by marcogalluzzi — APPROVED** [review]
Very good!

**Comment by marcogalluzzi** [comment]
ci:forward

**Comment by WYZ-Issac** [comment]
ci:test:relevant

**Comment by WYZ-Issac** [comment]
Failures are unrelated.

**Comment by WYZ-Issac** [comment]
ci:forward:force

**Pull request closed without merging**